groups/nts/ntsu/ntsu_socketutil.cpp

```diff
@@ -416,6 +421,138 @@ void OutoingDataStats::update(const struct iovec* buffersSendable,
 
 #endif
 
+#if NTSU_SOCKETUTIL_DEBUG_RECVMSG
+
+class IncomingDataStats
+{
+    // This class tracks the statistics for 'sendmsg'.
+
+    bsls::AtomicUint64 d_numSyscalls;
+    bsls::AtomicUint64 d_totalBuffersReceivable;
+    bsls::AtomicUint64 d_totalBytesReceivable;
+    bsls::AtomicUint64 d_totalBytesReceived;
+    bsls::AtomicUint64 d_numWouldBlocks;
+
+  public:
+    IncomingDataStats();
+
+    ~IncomingDataStats();
+
+    void update(const struct iovec* buffersReceivable,
+                bsl::size_t         numBuffersReceivable,
+                ssize_t             numBytesSent,
+                int                 errorNumber);
+};
+
+IncomingDataStats s_incomingDataStats;
+
+IncomingDataStats::IncomingDataStats()
+: d_numSyscalls(0)
+, d_totalBuffersReceivable(0)
+, d_totalBytesReceivable(0)
+, d_totalBytesReceived(0)
+, d_numWouldBlocks(0)
+{
+}
+
+IncomingDataStats::~IncomingDataStats()
+{
+    if (d_numSyscalls == 0) {
+        return;
+    }
+
+    double avgBuffersReceivablePerSyscall = 0;
+    if (d_numSyscalls > 0) {
+        avgBuffersReceivablePerSyscall =
+            static_cast<double>(d_totalBuffersReceivable) /
+            static_cast<double>(d_numSyscalls);
+    }
+
+    double avgBytesReceivablePerSyscall = 0;
+    if (d_numSyscalls > 0) {
+        avgBytesReceivablePerSyscall =
+            static_cast<double>(d_totalBytesReceivable) /
+            static_cast<double>(d_numSyscalls);
+    }
+
+    double avgBytesReceivedPerSyscall = 0;
+    if (d_numSyscalls > 0) {
+        avgBytesReceivedPerSyscall =
+                           static_cast<double>(d_totalBytesReceived) /
+                           static_cast<double>(d_numSyscalls);
+    }
+
+    char buffer[4096];
+
+    std::sprintf(buffer,
```

Comment:
Are we obliged to use `bsl` instead of `std` everywhere in the file?

Reply:
No, but let's be consistent and use `bsl` and not `std` unless there is some reason not to, which there isn't in this code. I'll change all usage of `std::` to `bsl::` in the code affected by this PR.